Compile form source: `make` didn't worked

### Steps <!-- how to reproduce the issue? -->
Install kakoune from source on ubuntu 19.04 disco 
After installing kakoune from package manager, I realised kakoune was not up to date ( see https://github.com/danr/kakoune-easymotion/issues/23) 
When I run `kakoune -version` I get `Kakoune unknown`
So I tried to install it from source, following the ubuntu tutorial on the github README

### Outcome <!-- what is the result of the above steps? -->
After updating, upgrading and installing the libraries, I run `make` in the `src` directory and got the following errors and warnings : 
```
 src/ git:(master) make
g++  -O3 -pedantic -std=c++17 -g -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-address -Wno-init-list-lifetime -MD -MP -MF .main.opt.d -c -o .main.opt.o main.cc
In file included from /usr/include/c++/8/vector:63,
                 from vector.hh:6,
                 from file.hh:9,
                 from buffer.hh:8,
                 from main.cc:3:
/usr/include/c++/8/bits/stl_uninitialized.h: In instantiation of ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, _Allocator&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _ForwardIterator = Kakoune::String*; _Allocator = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’:
/usr/include/c++/8/bits/stl_vector.h:1469:33:   required from ‘void std::vector<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:1437:4:   required from ‘void std::vector<_Tp, _Alloc>::_M_initialize_dispatch(_InputIterator, _InputIterator, std::__false_type) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:546:4:   required from ‘std::vector<_Tp, _Alloc>::vector(_InputIterator, _InputIterator, const allocator_type&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; <template-parameter-2-2> = void; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>; std::vector<_Tp, _Alloc>::allocator_type = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
ranges.hh:469:50:   required from ‘Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)> [with auto:10 = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >]’
ranges.hh:23:51:   required from ‘decltype(auto) Kakoune::operator|(Range&&, Kakoune::ViewFactory<Func>) [with Range = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >; Func = Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)>]’
main.cc:317:50:   required from here
/usr/include/c++/8/bits/stl_uninitialized.h:275:25: error: no matching function for call to ‘__gnu_cxx::__alloc_traits<Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>, Kakoune::String>::construct(Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>&, Kakoune::String*, Kakoune::InplaceString<15>)’
      __traits::construct(__alloc, std::__addressof(*__cur), *__first);
      ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/8/bits/basic_string.h:40,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/bits/locale_classes.h:40,
                 from /usr/include/c++/8/bits/ios_base.h:41,
                 from /usr/include/c++/8/ios:42,
                 from /usr/include/c++/8/ostream:38,
                 from /usr/include/c++/8/iterator:64,
                 from array_view.hh:5,
                 from constexpr_utils.hh:8,
                 from buffer.hh:6,
                 from main.cc:3:
/usr/include/c++/8/ext/alloc_traits.h:82:7: note: candidate: ‘template<class _Ptr, class ... _Args> static typename std::enable_if<std::__and_<std::is_same<typename std::allocator_traits<_Alloc>::pointer, _Ptr>, std::__not_<std::is_pointer<_Ptr> > >::value>::type __gnu_cxx::__alloc_traits<_Alloc, <template-parameter-1-2> >::construct(_Alloc&, _Ptr, _Args&& ...) [with _Ptr = _Ptr; _Args = {_Args ...}; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>; <template-parameter-1-2> = Kakoune::String]’
       construct(_Alloc& __a, _Ptr __p, _Args&&... __args)
       ^~~~~~~~~
/usr/include/c++/8/ext/alloc_traits.h:82:7: note:   template argument deduction/substitution failed:
/usr/include/c++/8/ext/alloc_traits.h: In substitution of ‘template<class _Ptr, class ... _Args> static typename std::enable_if<std::__and_<std::is_same<Kakoune::String*, _Ptr>, std::__not_<std::is_pointer<_Tp> > >::value, void>::type __gnu_cxx::__alloc_traits<Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>, Kakoune::String>::construct<_Ptr, _Args ...>(Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>&, _Ptr, _Args&& ...) [with _Ptr = Kakoune::String*; _Args = {Kakoune::InplaceString<15>}]’:
/usr/include/c++/8/bits/stl_uninitialized.h:275:25:   required from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, _Allocator&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _ForwardIterator = Kakoune::String*; _Allocator = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:1469:33:   required from ‘void std::vector<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:1437:4:   required from ‘void std::vector<_Tp, _Alloc>::_M_initialize_dispatch(_InputIterator, _InputIterator, std::__false_type) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:546:4:   required from ‘std::vector<_Tp, _Alloc>::vector(_InputIterator, _InputIterator, const allocator_type&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; <template-parameter-2-2> = void; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>; std::vector<_Tp, _Alloc>::allocator_type = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
ranges.hh:469:50:   required from ‘Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)> [with auto:10 = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >]’
ranges.hh:23:51:   required from ‘decltype(auto) Kakoune::operator|(Range&&, Kakoune::ViewFactory<Func>) [with Range = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >; Func = Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)>]’
main.cc:317:50:   required from here
/usr/include/c++/8/ext/alloc_traits.h:82:7: error: no type named ‘type’ in ‘struct std::enable_if<false, void>’
In file included from /usr/include/c++/8/ext/alloc_traits.h:36,
                 from /usr/include/c++/8/bits/basic_string.h:40,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/bits/locale_classes.h:40,
                 from /usr/include/c++/8/bits/ios_base.h:41,
                 from /usr/include/c++/8/ios:42,
                 from /usr/include/c++/8/ostream:38,
                 from /usr/include/c++/8/iterator:64,
                 from array_view.hh:5,
                 from constexpr_utils.hh:8,
                 from buffer.hh:6,
                 from main.cc:3:
/usr/include/c++/8/bits/stl_uninitialized.h: In instantiation of ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, _Allocator&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _ForwardIterator = Kakoune::String*; _Allocator = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’:
/usr/include/c++/8/bits/stl_vector.h:1469:33:   required from ‘void std::vector<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:1437:4:   required from ‘void std::vector<_Tp, _Alloc>::_M_initialize_dispatch(_InputIterator, _InputIterator, std::__false_type) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:546:4:   required from ‘std::vector<_Tp, _Alloc>::vector(_InputIterator, _InputIterator, const allocator_type&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; <template-parameter-2-2> = void; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>; std::vector<_Tp, _Alloc>::allocator_type = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
ranges.hh:469:50:   required from ‘Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)> [with auto:10 = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >]’
ranges.hh:23:51:   required from ‘decltype(auto) Kakoune::operator|(Range&&, Kakoune::ViewFactory<Func>) [with Range = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >; Func = Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)>]’
main.cc:317:50:   required from here
/usr/include/c++/8/bits/alloc_traits.h:342:14: note: candidate: ‘template<class _Tp, class ... _Args> static decltype (std::allocator_traits<_Alloc>::_S_construct(__a, __p, (forward<_Args>)(std::allocator_traits::construct::__args)...)) std::allocator_traits<_Alloc>::construct(_Alloc&, _Tp*, _Args&& ...) [with _Tp = _Tp; _Args = {_Args ...}; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
  static auto construct(_Alloc& __a, _Tp* __p, _Args&&... __args)
              ^~~~~~~~~
/usr/include/c++/8/bits/alloc_traits.h:342:14: note:   template argument deduction/substitution failed:
/usr/include/c++/8/bits/alloc_traits.h: In substitution of ‘template<class _Tp, class ... _Args> static decltype (std::allocator_traits<Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >::_S_construct(__a, __p, (forward<_Args>)(std::allocator_traits::construct::__args)...)) std::allocator_traits<Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >::construct<_Tp, _Args ...>(Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>&, _Tp*, _Args&& ...) [with _Tp = Kakoune::String; _Args = {Kakoune::InplaceString<15>}]’:
/usr/include/c++/8/bits/stl_uninitialized.h:275:25:   required from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, _Allocator&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _ForwardIterator = Kakoune::String*; _Allocator = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:1469:33:   required from ‘void std::vector<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:1437:4:   required from ‘void std::vector<_Tp, _Alloc>::_M_initialize_dispatch(_InputIterator, _InputIterator, std::__false_type) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
/usr/include/c++/8/bits/stl_vector.h:546:4:   required from ‘std::vector<_Tp, _Alloc>::vector(_InputIterator, _InputIterator, const allocator_type&) [with _InputIterator = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator; <template-parameter-2-2> = void; _Tp = Kakoune::String; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>; std::vector<_Tp, _Alloc>::allocator_type = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
ranges.hh:469:50:   required from ‘Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)> [with auto:10 = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >]’
ranges.hh:23:51:   required from ‘decltype(auto) Kakoune::operator|(Range&&, Kakoune::ViewFactory<Func>) [with Range = Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >; Func = Kakoune::gather() [with Container = std::vector<Kakoune::String, Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >]::<lambda(auto:10&&)>]’
main.cc:317:50:   required from here
/usr/include/c++/8/bits/alloc_traits.h:343:26: error: no matching function for call to ‘std::allocator_traits<Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0> >::_S_construct(Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>&, Kakoune::String*&, Kakoune::InplaceString<15>)’
  -> decltype(_S_construct(__a, __p, std::forward<_Args>(__args)...))
              ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/8/bits/alloc_traits.h:242:2: note: candidate: ‘template<class _Tp, class ... _Args> static std::_Require<typename std::allocator_traits<_Alloc>::__construct_helper<_Tp, _Args>::type> std::allocator_traits<_Alloc>::_S_construct(_Alloc&, _Tp*, _Args&& ...) [with _Tp = _Tp; _Args = {_Args ...}; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
  _S_construct(_Alloc& __a, _Tp* __p, _Args&&... __args)
  ^~~~~~~~~~~~
/usr/include/c++/8/bits/alloc_traits.h:242:2: note:   template argument deduction/substitution failed:
/usr/include/c++/8/bits/alloc_traits.h:249:2: note: candidate: ‘template<class _Tp, class ... _Args> static std::_Require<std::__and_<std::__not_<typename std::allocator_traits<_Alloc>::__construct_helper<_Tp, _Args>::type>, std::is_constructible<_Tp, _Args ...> > > std::allocator_traits<_Alloc>::_S_construct(_Alloc&, _Tp*, _Args&& ...) [with _Tp = _Tp; _Args = {_Args ...}; _Alloc = Kakoune::Allocator<Kakoune::String, (Kakoune::MemoryDomain)0>]’
  _S_construct(_Alloc&, _Tp* __p, _Args&&... __args)
  ^~~~~~~~~~~~
/usr/include/c++/8/bits/alloc_traits.h:249:2: note:   template argument deduction/substitution failed:
In file included from unicode.hh:8,
                 from utf8.hh:5,
                 from string.hh:7,
                 from enum.hh:4,
                 from buffer.hh:7,
                 from main.cc:3:
ranges.hh:192:32: warning: inline function ‘Kakoune::TransformView<Kakoune::BufferManager&, Kakoune::transform(M T::*) [with M = const Kakoune::String&() const; T = Kakoune::Buffer]::<lambda(auto:6&&)> >::Iterator::difference_type Kakoune::operator-(const Kakoune::TransformView<Kakoune::BufferManager&, Kakoune::transform(M T::*) [with M = const Kakoune::String&() const; T = Kakoune::Buffer]::<lambda(auto:6&&)> >::Iterator&, const Kakoune::TransformView<Kakoune::BufferManager&, Kakoune::transform(M T::*) [with M = const Kakoune::String&() const; T = Kakoune::Buffer]::<lambda(auto:6&&)> >::Iterator&)’ used but never defined
         friend difference_type operator-(const Iterator& lhs, const Iterator& rhs) { return lhs.m_it - rhs.m_it; }
                                ^~~~~~~~
ranges.hh:192:32: warning: inline function ‘Kakoune::TransformView<Kakoune::ClientManager&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const std::unique_ptr<Kakoune::Client>&)> >::Iterator::difference_type Kakoune::operator-(const Kakoune::TransformView<Kakoune::ClientManager&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const std::unique_ptr<Kakoune::Client>&)> >::Iterator&, const Kakoune::TransformView<Kakoune::ClientManager&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const std::unique_ptr<Kakoune::Client>&)> >::Iterator&)’ used but never defined
ranges.hh:192:32: warning: inline function ‘Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator::difference_type Kakoune::operator-(const Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator&, const Kakoune::TransformView<const Kakoune::SelectionList&, Kakoune::<lambda(Kakoune::StringView, const Kakoune::Context&)>::<lambda(const Kakoune::Selection&)> >::Iterator&)’ used but never defined
cc1plus: warning: unrecognized command line option ‘-Wno-init-list-lifetime’
make: *** [Makefile:113: .main.opt.o] Error 1
```



### Expected <!-- what should have been the result of the above steps? -->
Compile the code successfully

--- a/src/main.cc
+++ b/src/main.cc
@@ -312,7 +312,7 @@ static const EnvVarDesc builtin_env_vars[] = { {
         "selections_length", false,
         [](StringView name, const Context& context) -> Vector<String>
         { return context.selections() |
-                     transform([&](const Selection& s) {
+                     transform([&](const Selection& s) -> String {
                          return to_string(char_length(context.buffer(), s));
                      }) | gather<Vector<String>>(); }
     }, {

